### ISSUE TITLE
CLI can't work with existing project without node_modules

### ISSUE THREAD

**Issue by kschoenrock-iw** [serverless/serverless#7580]
I have created a serverless project for python on AWS, which works fine... until I try to re-use the project directly after checkout. Any CLI command exits with the following error:
```
 sls help
 
 Serverless Error ---------------------------------------
 
  Serverless plugin "serverless-python-requirements" not found. Make sure it's installed and listed in the "plugins" section of your serverless config file.
 
  Get Support --------------------------------------------
     Docs:          docs.serverless.com
     Bugs:          github.com/serverless/serverless/issues
     Issues:        forum.serverless.com
 
  Your Environment Information ---------------------------
     Operating System:          linux
     Node Version:              12.16.1
     Framework Version:         1.67.3 (standalone)
     Plugin Version:            3.6.6
     SDK Version:               2.3.0
     Components Version:        2.29.0
```
I installed Serverless with the method "Install as a standalone binary". The node_modules folder wasn't checked in, but package.json and package-lock.json are versioned with serverless.yml, and the plugin is also present in serverless.yml. Is this a bug, or am I missing some initialization step?

**Comment by JeremiGendron** [comment]
The error you're getting is because serverless is trying to find serverless-python-requirements in your node_modules but it isn't there.

When you checkout, none of your dependencies are installed in the folder. The `sls` function isn't responsible for installing your dependencies. You need to re-install them by running `npm install` once so that `sls` can use your dependencies.

This is standard for all NodeJS projects, when you clone or checkout a branch you should npm install so that the dependencies are downloaded into node_modules.

**Comment by kschoenrock-iw** [comment]
Yes, I know now that `npm install` is necessary, but following the "standalone binary" installation results in a working Serverless without npm, which leaves people scratching their heads when this step is needed.
Some extra documentation would be very useful for this (I had touched npm perhaps twice in my life before using serverless, and it wasn't immediately obvious what I had to do in this situation). Since serverless itself already worked, and knowing it was based on NodeJS, I had expected it to have bundled its own NodeJS tools in the "standalone binary" installation.

**Comment by medikoo** [comment]
> Yes, I know now that npm install is necessary, but following the "standalone binary" installation results in a working Serverless without npm, which leaves people scratching their heads when this step is needed.

If you do not have npm installed, you may install plugins via `sls plugin install <plugin-name>` then npm that's bundled with sls binary will be used for plugin installation

**Comment by kschoenrock-iw** [comment]
@medikoo I did install the plugin using `sls plugin install`, which is why I assumed there would be a command to reinstall plugins à la `npm install`, since npm is already bundled with the standalone installation.
IMO this would be a nice option to have.

**Comment by medikoo** [comment]
@kschoenrock-iw so issue is that you're not sure how to update (e.g. upgrade to more recent version) an already installed plugin?

**Comment by kschoenrock-iw** [comment]
No, I'd say it was just missing documentation for people less familiar with NodeJS using the Serverless standalone installation method. 

The best solution would be taking advantage of the bundled `npm` to automate such setup and management tasks from `sls`.

**Comment by medikoo** [comment]
@skierkowski I remember we were discussing documenting plugin installation as `sls plugin install <plugin-name>` instead of `npm install <plugin-name>`, do we plan to go that route?

It looks it's valuable especially to _standalone_ users (and harmless for those relying on npm installation)

**Comment by Alynva** [comment]
Could `sls plugin install` be an alias to `npm install`?

**Comment by medikoo** [comment]
@Alynva Those are very different commands

**Comment by Alynva** [comment]
> @Alynva Those are very different commands

So how do you solve the problem of people that is not NodeJs Dev, installed the binary version of Serverless and needs to setup their own environment?

**Comment by medikoo** [comment]
> So how do you solve the problem of people that is not NodeJs Dev, installed the binary version of Serverless and needs to setup their own environment?

What exactly is the problem?

**Comment by Alynva** [comment]
> > So how do you solve the problem of people that is not NodeJs Dev, installed the binary version of Serverless and needs to setup their own environment?
> 
> What exactly is the problem?

When they clone, they need to install the plugins locally. That would be done with `npm install`, but they don't have `npm` installed as they installed the binary version of Serverless. Instead, they have to run `serverless plugin install <plugin-name>` for each plugin in `serverless.yml`.

Would be better to just have a single command that installs all necessary plugins, as `npm install` or `npm ci` does.

**Comment by medikoo** [comment]
> Would be better to just have a single command that installs all necessary plugins, as npm install or npm ci does.

I see, yes indeed there's no method that will just install all plugin dependencies for an existing service. Interestingly it's covered when you set up a new project via the `serverless` command and choose one of the existing templates. 

I guess `sls install` could do it, if we make `url` option as optional WDYT @pgrzesik @mnapoli ?

**Comment by pgrzesik** [comment]
That's an interesting proposition, but I know we've discussed deprecating `sls install` in the future. What do you think @mnapoli ?

**Comment by medikoo** [comment]
> but I know we've discussed deprecating sls install in the future

Yes, I believe it should still be deprecated for what it does currently.  

Eventually instead of updating `sls install` to cover that use case. We may cover it in `serverless`. It can detect if all npm dependencies for the service are installed and if not propose to install them.

**Comment by mnapoli** [comment]
We want to deprecate `sls install` in favor of the main onboarding. But we don't want to deprecate `sls plugin install`.

> Could `sls plugin install` be an alias to `npm install`/`npm ci`?

So that suggestion from @Alynva is a good idea:

```sh
# Install a plugin
sls plugin install -n <plugin>

# Install all referenced plugins (i.e. runs `npm install`)
sls plugin install
```

However something that could be confusing: `sls plugin install` would run `npm install`, which would install all dependencies from `package.json` (not just the plugins).

Is that what you meant @Alynva?

**Comment by Alynva** [comment]
> Is that what you meant @Alynva?

Exactly. I was already thinking about running the "global install", but if the output is clear enough, I guess it's ok.

Also, just tested, if you add some plugin A by editing the `package.json` and then install another plugin B with the `sls plugin install -n <name>`, the plugin A, that was not installed (not in `package-lock.json` nor in `node_modules`), is also installed with the plugin B.

So the behavior of `sls plugin install` would be the same...
